test/suite/test_bug031.py

```diff
@@ -0,0 +1,152 @@
+    format_values = [
+        ('column', dict(key_format='r', value_format='S')),
+        ('column_fix', dict(key_format='r', value_format='8t')),
+        ('row_integer', dict(key_format='i', value_format='S')),
+    ]
+
+    scenarios = make_scenarios(format_values)
+
+    def conn_config(self):
+        config = 'debug_mode=(update_restore_evict=true)'
+        return config
+
+    def test_bug031(self):
+        uri = "table:test_bug031"
+
+        key = 1
+        if self.value_format == '8t':
+            value = 1
+        else:
+            value = "1"
+
+        self.session.create(uri, 'key_format={},value_format={}'.format(
+            self.key_format, self.value_format))
+        cursor = self.session.open_cursor(uri)
+        
+        # Perform a first insertion.
+        self.session.begin_transaction()
+        cursor[key] = value
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(10))
+
+        # Set the stable and oldest timestamps.
+        self.conn.set_timestamp('oldest_timestamp={},stable_timestamp={}'.format(
+            self.timestamp_str(1), self.timestamp_str(10)))
+
+        # Remove the record.
+        self.session.begin_transaction()
+        cursor.set_key(key)
+        cursor.remove()
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(20))
+
+        # Evict everything.
+        evict_cursor = self.session.open_cursor(uri, None, 'debug=(release_evict)')
+        evict_cursor.set_key(key)
+        evict_cursor.search()
+        evict_cursor.reset()
+        evict_cursor.close()
+
+        # After eviction, we have:
+        # - Nothing in memory.
+        # - A record with a start ts 10 and stop ts 20 in the DS.
+        # - Nothing in the HS.
+
+        # Insert an update.
+        self.session.begin_transaction()
+        cursor[key] = value
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(30))
+
+        # Checkpoint.
+        self.session.checkpoint()
+
+        # After the checkpoint, we expect the following:
+        # - In memory, an update list with insert @ 30 (U3), remove @ 20 (U2) and insert @ 10 (U1).
+        # The two updates U1 and U2 should have the flag WT_UPDATE_PREPARE_RESTORED_FROM_DS as they
+        # were read back into memory from disk by the checkpoint.
+        # - The most recent insertion U3 in the DS.
+        # - A record with a start ts 10 and stop ts 20 in the HS.
+
+        # RTS.
+        self.conn.rollback_to_stable()
+
+        # Since the stable timestamp is 10, RTS marks all the updates performed at a time greater
+        # time than 10 as aborted. The update chain in memory is:
+        # U3 (aborted) -> U2 (aborted) -> U1.
+        # The DS now contains the insert @ 10 (U1) and the HS is empty.
+
+        # Insert but don't commit yet.
+        self.session.begin_transaction()
+        cursor[key] = value
+
+        # Evict everything.
```

Comment:
How about adding more details like this eviction performs update restore evict due to the uncommitted update present on the key?

Reply:
Added in 183179f07305e57a80978edf887c5a1e28243a40.